### PR TITLE
Implemented Initial Meal Sharing API Endpoints

### PR DESCRIPTION
**Pull request by sumankaundinya**
This PR adds the initial set of backend API endpoints for the Meal Sharing project using Express and Knex. The following routes are now available:-
1)GET /future-meals – Returns meals scheduled in the future.
2)GET /past-meals – Returns meals that occurred in the past.
3)GET /all-meals – Returns all meals sorted by ID.
4)GET /first-meal – Returns the meal with the lowest ID.
5)GET /last-meal – Returns the meal with the highest ID.

### CODE REVIEW

File: api/src/index.js [hunk 1]
```diff
@@ -2,30 +2,82 @@ import "dotenv/config";
...
+
+// Routes
+app.get("/my-route", (req, res) => {
+  res.send("Hi friend");
+});
+
+app.get("/future-meals", async (req, res) => {
+  try {
+    const meals = await knex.raw("SELECT * FROM Meal WHERE `when` > NOW()");
+    res.json(meals[0]);
```
Comment (aman-shahid) [inline api/src/index.js]: what if meals is empty or undefined?

File: api/src/index.js [hunk 2]
```diff
@@ -2,30 +2,82 @@ import "dotenv/config";
...
+    res.json(meals[0][0]);
+  } catch (err) {
+    res.status(500).json({ error: err.message });
+  }
+});
+
+app.get("/last-meal", async (req, res) => {
+  try {
+    const meals = await knex.raw("SELECT * FROM Meal ORDER BY id DESC LIMIT 1");
+    if (meals[0].length === 0) {
```
Comment (aman-shahid) [inline api/src/index.js]: hmmm, what if meals is empty/undefined?

File: api/src/index.js [hunk 3]
```diff
@@ -2,30 +2,82 @@ import "dotenv/config";
...
 import cors from "cors";
 import bodyParser from "body-parser";
-import knex from "./database_client.js";
-import nestedRouter from "./routers/nested.js";
+import knexLib from "knex";
 
 const app = express();
+const port = process.env.PORT || 3000;
+
 app.use(cors());
```
Comment (aman-shahid) [inline api/src/index.js:10]: good job on adding cors!

File: api/src/index.js [hunk 4]
```diff
@@ -2,30 +2,82 @@ import "dotenv/config";
...
+});
 
-app.use("/api", apiRouter);
+app.get("/first-meal", async (req, res) => {
+  try {
+    const meals = await knex.raw("SELECT * FROM Meal ORDER BY id ASC LIMIT 1");
+    if (meals[0].length === 0) {
+      return res.status(404).json({ message: "No meals found" });
+    }
+    res.json(meals[0][0]);
```
Comment (aman-shahid) [inline api/src/index.js]: Does knex.raw return a nested array ?

**Review by aman-shahid — COMMENTED** [review]
Good work on adding routes for all endpoints. 
Please address some of the coments and I will get back to your PR.

Btw, have you tried running the code to test some edge cases like empty meals table etc. ?

File: api/src/index.js (hunk 1 above)
Comment (sumankaundinya) [reply api/src/index.js]: Hi Aman!
 
 If the query finds nothing, meals[0] will be an empty array and the output will be:
 [ ] 
 and can be handled with a message instead of an empty array [ ] like this "if (!meals || meals.length === 0) {
      return res.status(404).json({ message: "No meals found" });
    }"

File: api/src/index.js (hunk 2 above)
Comment (sumankaundinya) [reply api/src/index.js]: Hi Aman!

If the query finds nothing, meals[0] will be an empty array and the output will be:
[ ]
and can be handled with a message instead of an empty array [ ] like this "if (!meals || meals.length === 0) {
return res.status(404).json({ message: "No meals found" });
}"

File: api/src/index.js (hunk 4 above)
Comment (sumankaundinya) [reply api/src/index.js]: Yes,
it returns something like this
 [
  [ 
    { id: 1, title: 'Pizza', when_date: '2025-05-17', ... }  
  ],
  metadata 
]
and meals[0][0] extracts only the object required instead of array where meals[0] extracts the array of objects even for a single row required.

**Comment by sumankaundinya** [comment]
> Good work on adding routes for all endpoints. Please address some of the coments and I will get back to your PR.
> 
> Btw, have you tried running the code to test some edge cases like empty meals table etc. ?

Thank you Aman!

Got these results for example : 
1){"error":"SELECT * FROM Meal ORDER BY id ASC LIMIT 1 - Unknown database 'meal_sharing'"}--if there is no database 

2){"message":"No meals found"}--if the meal table is empty
